src/test/java/it/geosolutions/httpproxy/HttpProxyTest.java

```diff
@@ -20,96 +20,261 @@
+        when(getRequest.getRequestURL()).thenReturn(
+                new StringBuffer("http://proxy.com/http-proxy/proxy"));
+        // mock http response object
+        HttpServletResponse getResponse = mock(HttpServletResponse.class);
+        final StubServletOutputStream servletOutputStream = new StubServletOutputStream();
+        when(getResponse.getOutputStream()).thenReturn(servletOutputStream);
+
+        proxy.doGet(getRequest, getResponse);
+        verify(getResponse).sendRedirect(
+                "http://proxy.com/http-proxy/proxy?url="
+                        + URLEncoder.encode(fakeLocation, "UTF-8"));
+        final byte[] data = servletOutputStream.baos.toByteArray();
+        Assert.assertNotNull(data);
+        Assert.assertTrue(data.length == 0);
+    }
+
+    @Test
+    public void testPost() throws Exception {
+
+        // mock post response
+        final HttpPost mockPostMethod = mock(HttpPost.class);
+        HttpResponse response = mock(HttpResponse.class);
+        StatusLine statusLine = mock(StatusLine.class);
+        when(statusLine.getStatusCode()).thenReturn(200);
+        when(response.getStatusLine()).thenReturn(statusLine);
+
+        HttpEntity stringEntity = new StringEntity("user created");
+        when(response.getEntity()).thenReturn(stringEntity);
+        when(mockPostMethod.getAllHeaders()).thenReturn(new Header[]{});
+        mockHttpClient = mock(HttpClient.class);
+        when(mockHttpClient.execute(mockPostMethod)).thenReturn(response);
+
+        proxy = new HTTPProxy() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public HttpPost getPostMethod(URL url) {
+                return mockPostMethod;
+            }
+        };
+        proxy.setHttpClient(mockHttpClient);
+        proxy.init(servletConfig);
+
+        // mock http request to proxy
+        HttpServletRequest postRequest = mock(HttpServletRequest.class);
+        when(postRequest.getQueryString()).thenReturn("url=https://jsonplaceholder.typicode.com/test/createUser");
+        when(postRequest.getMethod()).thenReturn("post");
+
+        ServletInputStream stream = mock(ServletInputStream.class);
+        when(postRequest.getInputStream()).thenReturn(stream);
+
+        // mock http response object
+        HttpServletResponse getResponse = mock(HttpServletResponse.class);
+        final StubServletOutputStream servletOutputStream = new StubServletOutputStream();
+        when(getResponse.getOutputStream()).thenReturn(servletOutputStream);
+
+        proxy.doPost(postRequest, getResponse);
+        verify(getResponse).setStatus(200);
+        final byte[] data = servletOutputStream.baos.toByteArray();
+        Assert.assertNotNull(data);
+        Assert.assertTrue(data.length != 0);
+    }
+
+    /**
+     * Run the local http proxy server to test and pass below Wiremock tests
+     */
+
+    @Test
+    public void testGETUsingWireMock() throws IOException {
```

Comment:
These tests are indeed more realistic than the full mocked ones, well done. I would recommend creating a new test class named ``HttpProxyIntegrationTests``.